### ISSUE TITLE
Found multiple accounts with same username. 

### ISSUE THREAD

**Issue by dylankdreyer78** [Azure/azure-cli#20168]
When trying to log in using the CLI I get this error:

![image](https://user-images.githubusercontent.com/93645505/140031297-c84248ad-8429-4419-910a-bdb3c9cca3e2.png)

I have obviously made a mess of things here but I don't know how to sort this out now :/ 

Any ideas?

**Comment by jiasli** [comment]
Could you please share some extra information:

1. How is Azure CLI installed? Did you install it following https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-windows?tabs=azure-cli or using `pip`?
2. Could you turn off token encryption first and log in again?
    ```
    az config set core.token_encryption=false
    az login
    ```
    Then share the content of `Account` property in `C:\Users\<username>\.azure\msal_token_cache.json`.

**Comment by dylankdreyer78** [comment]
Thanks so much for pointing me in the right direction. cleared the cache and works 100% now!

**Comment by jiasli** [comment]
@dylankdreyer78, I am glad to know it works for you. Do you mind letting us know what action you have performed? Thanks a lot.

**Comment by dylankdreyer78** [comment]
No problem. I deleted the json files in the .azure directory :)

**Comment by jiasli** [comment]
Hmm... This still didn't reveal why it happened in the first place. Anyway, we will keep monitoring for other occurrences. Thanks for letting us know.

**Comment by hivyas** [comment]
Hi there, I am seeing the same issue. Deleting the cache file and relogging in did not work. I also only see a `msal_token_cache.bin` file, not a .json file.

**Comment by jiasli** [comment]
@hivyas, as shown in the output, these 2 accounts share the same `"home_account_id": "508c96e0-d4ff-4c30-b886-34c377353339.72f988bf-86f1-41af-91ab-2d7cd011db47"`. In such case, MSAL will group them (https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/349). 

Are you using Azure CLI from source code? If so, please verify if your MSAL is update to date and configure all Azure CLI dependencies again.

```
pip list
azdev setup --cli
```

**Comment by Tomnowell** [comment]
Hi jiasli, Thanks for the email.  I'm in a similar situation.  

I Installed Azure CLI from .msi package 

Heres contents of unencrypted account tokens

`"Account": {
        "00000000-0000-0000-7646-451e2554e8a2.9188040d-6c67-4c5b-b112-36a304b66dad-login.microsoftonline.com-organizations": {
            "home_account_id": "00000000-0000-0000-7646-451e2554e8a2.9188040d-6c67-4c5b-b112-36a304b66dad",
            "environment": "login.microsoftonline.com",
            "realm": "organizations",
            "local_account_id": "mxZ7AZSUax9B16n3lwE4isegKGToyJ0ZIK1iDkfY5vQ",
            "username": "<redacted>",
            "authority_type": "MSSTS"
        },
        "237e830d-aa10-4fce-bdbc-62bf9acc8b73.98c2b1ec-775b-49fa-9ba5-1362cc916655-login.microsoftonline.com-organizations": {
            "home_account_id": "237e830d-aa10-4fce-bdbc-62bf9acc8b73.98c2b1ec-775b-49fa-9ba5-1362cc916655",
            "environment": "login.microsoftonline.com",
            "realm": "organizations",
            "local_account_id": "237e830d-aa10-4fce-bdbc-62bf9acc8b73",
            "username": "<redacted>",
            "authority_type": "MSSTS"
        }
    },`

thanks

**Comment by jiasli** [comment]
Is the `<redacted>` username the same? It is uncommon for the same `<redacted>` username to have 2 district `home_account_id`s. +@rayluo

**Comment by Tomnowell** [comment]
Sorry, yes the username is the same. It's my private email address.  When I log in via a browser I am asked which account I want:  the one set up by me, or one set up by my IT department.  As I don't have an IT department I'm not sure why they are separate.  I tried to "merge" the accounts but that hasn't solved the problem.

**Comment by jiasli** [comment]
Let me involve MSAL developer and AAD expert @rayluo to help take a look.

**Comment by jiasli** [comment]
After discussing with MSAL developer @rayluo, we feel this is an unexpected scenario. `9188040d-6c67-4c5b-b112-36a304b66dad` is the tenant ID for Microsoft Account.

I tried to log in with my own Microsoft Account and the `Account` field looks like:

```
    "Account": {
        "00000000-0000-0000-a399-d570a52cd9de.9188040d-6c67-4c5b-b112-36a304b66dad-login.microsoftonline.com-organizations": {
            "home_account_id": "00000000-0000-0000-a399-d570a52cd9de.9188040d-6c67-4c5b-b112-36a304b66dad",
            "environment": "login.microsoftonline.com",
            "realm": "organizations",
            "local_account_id": "hfwAXfMGhps2MGbpc70lxh7wAMljXr6y1wqiSvvMfnw",
            "username": "myname@outlook.com",
            "authority_type": "MSSTS"
        }
    },
```

However, in your case, the same email address is also an AAD account, because it has a full `"home_account_id": "237e830d-aa10-4fce-bdbc-62bf9acc8b73.98c2b1ec-775b-49fa-9ba5-1362cc916655"`.

For now, the solution would be to clear the Azure CLI storage and MSAL token cache, then re-login with **only one account**:

```
az account clear
az login
```

I believe this is how @dylankdreyer78 solved this issue.

To better help us understand your account configuration, would you mind sending a screenshot of your AAD login page to my email address jiasli@microsoft.com? We can then help you on this issue privately. 

Thanks a lot for the information.

**Comment by spottedmahn** [comment]
>Do you mind letting us know what action you have performed?

I got a personal & work account w/ the same email / username... I logged in w/ one & get that message on `az provider register --namespace Microsoft.Web` 🤷‍♂️

I've emailed you my powershell session, if that helps.

**Comment by jiasli** [comment]
> I got a personal & work account w/ the same email / username...

I am not sure if this due to some historical reason, but currently I am not able to create **Microsoft Account** with **work or school account email**:

![image](https://user-images.githubusercontent.com/4003950/140672927-54bfe9c0-e458-47a5-915e-ba90c66f63b5.png)

`@microsoft.com` is also reserved.

![image](https://user-images.githubusercontent.com/4003950/140672956-1c6d4ee7-2ea8-4f82-a37c-635cae9f5404.png)

**Comment by jiasli** [comment]
@spottedmahn, thanks for the email. Apparently, `az account clear` and `az login` have helped solve the "Found multiple accounts with the same username" issue.

However, your last problem doesn't seem to be related to this issue, it is rather a problem with `az provider register`:

```
> az provider register --namespace Microsoft.Web
Registering is still on-going. You can monitor using 'az provider show -n Microsoft.Web'
```

As the error suggests, the registration is on-going and it should complete in a while. If you still see any problem with this command, could you create a new issue? Thanks.

**Comment by spottedmahn** [comment]
>However, your last problem doesn't seem to be related to this issue

I figured not, thanks 👍

**Comment by jiasli** [comment]
## Solution

To get unblocked, please clear Azure CLI's accounts and log in again with only one account (Microsoft Account or AAD Account):

```
az account clear
az login
```

Later if you want to logging in again, please make sure to select the same account, but not the other one.

**Comment by jiasli** [comment]
A little more context:

Previously, ADAL raises an exception when multiple tokens are found for the same username:

https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/1.2.7/adal/cache_driver.py#L123-L124

```py
                raise AdalError('More than one token matches the criteria. The result is ambiguous.')
```

This was reported at #5548, #6147, #6957, #15320 ([search link](https://github.com/Azure/azure-cli/issues?q=is%3Aissue++is%3Aclosed+%22More+than+one+token+matches+the+criteria.+The+result+is+ambiguous.%22)).

**Comment by CitypassengerIT** [comment]
Locked

**Comment by jiasli** [comment]
@CitypassengerIT, `"state": "Enabled"` is from `~/.azure/azureProfile.json` indicating the subscription is enabled. It has nothing to do with the MSAL token cache `~/.azure/msal_token_cache.bin` (encrypted) or `~/.azure/msal_token_cache.json` which is the direct cause of this "multiple accounts" error.

**Comment by jiasli** [comment]
According to a [search on issues](https://github.com/Azure/azure-cli/issues?q=is%3Aissue+is%3Aopen+sandbox+multiple+accounts), several issues are caused by sandbox account:

- #20364 
- #20342
- #20317

Found these resources for **Microsoft Learn sandbox**:

- https://docs.microsoft.com/en-us/learn/support/faq?pivots=sandbox
- https://community.dynamics.com/crm/b/dynamics-365-blog-with-manish/posts/learn-azure-with-a-free-sandbox-subscription

We will do some investigation about this topic.

**Comment by johanchen** [comment]
> @dylankdreyer78, I am glad to know it works for you. Do you mind letting us know what action you have performed? Thanks a lot.

For me, this below command solved the same issue highlighted by @dylankdreyer78
az config set core.token_encryption=false

**Comment by rayluo** [comment]
> > @dylankdreyer78, I am glad to know it works for you. Do you mind letting us know what action you have performed? Thanks a lot.
> 
> For me, this below command solved the same issue highlighted by @dylankdreyer78 az config set core.token_encryption=false

@johanchen , I would assume your statement above was not precise. The situation we know so far is:
* The workaround to the "Found multiple accounts with same username" symptom is to do `az account clear` and then you can `az login` again. But we do not yet fully understand how that symptom would happen in the first place.
* The `az config set core.token_encryption=false` is an attempt to temporarily disable token encryption, for you to subsequently share the content of `Account` property in `C:\Users\<username>\.azure\msal_token_cache.json` to us, to help our investigation. Turning the encryption on or off would not change the "multiple accounts" situation.

**Comment by deeTEEcee** [comment]
If it helps, I ran into this too and this was because during login, I have two accounts registered to the same email: one was the work account and one was identified as personal account. The latter I might've made just while being confused by the process of sync'ing my email with our azure stuff. 

So I would do `az login` with the personal email and that one would fail. I would attempt it again and use the work one but then run into this issue.

**Comment by vid** [comment]
I see this message on Ubuntu Linux, though I can continue with the flow. Happy to help diagnose from this state. I think my account is a mess behind the scenes, a few things don't work as I expect. For example, unrelated to this, I can't properly log into a devops project using github, which seems to prevent me from inviting other github contributors to that repo, which is a problem I have no idea how to sort out.

**Comment by vid** [comment]
And now I can't continue, so I am blocked from doing my work, again.

**Comment by mcclung** [comment]
Last night I was using microsoft learn to create a sandbox.  The sandbox creation failed initially, but I was left with an account that is not signed into my normal PAYG subscription.    At some point I visited aad.portal.azure.com and during the log in process things started to go wrong.  The CLI (version I installed with Gentoo Linux) gave me this:

```
$ az login
The default web browser has been opened at https://login.microsoftonline.com/organizations/oauth2/v2.0/authorize. Please continue the login in the web browser. If no web browser is available or if the web browser fails to open, use device code flow with `az login --use-device-code`.
Opening in existing browser session.
[19176:19176:0100/000000.166713:ERROR:vaapi_wrapper.cc(1119)] vaQuerySurfaceAttributes failed, VA error: invalid parameter
[19176:19176:0100/000000.166756:ERROR:vaapi_wrapper.cc(1066)] FillProfileInfo_Locked failed for va_profile VAProfileH264Main and entrypoint VAEntrypointVLD
[19176:19176:0100/000000.166786:ERROR:vaapi_wrapper.cc(1119)] vaQuerySurfaceAttributes failed, VA error: invalid parameter
[19176:19176:0100/000000.166809:ERROR:vaapi_wrapper.cc(1066)] FillProfileInfo_Locked failed for va_profile VAProfileH264High and entrypoint VAEntrypointVLD
Found multiple accounts with the same username 'scott@mcclung.com': [{'home_account_id': '00000000-0000-0000-baf4-282f003203cd.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'scott@mcclung.com', 'authority_type': 'MSSTS', 'local_account_id': 'eRI7YFbzT2A78sky0As1Ix6SDHNqwM3-satLWHHKM5c', 'realm': 'organizations'}, {'home_account_id': '35d7b220-2b2d-4b82-a5a1-2e8f2907fd79.a9bc815e-e46f-4c95-9c96-bc46c4d342b5', 'environment': 'login.microsoftonline.com', 'username': 'scott@mcclung.com', 'authority_type': 'MSSTS', 'local_account_id': '35d7b220-2b2d-4b82-a5a1-2e8f2907fd79', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168


```
I deleted ~/.azure and started over and got the right login this time:

```
[
  {
    "cloudName": "AzureCloud",
    "homeTenantId": "5b234a67-1ad7-4021-9056-328b95f63810",
    "id": "e8868903-7f75-40d3-8cf6-9ac1a3fbd5c1",
    "isDefault": true,
    "managedByTenants": [],
    "name": "Azure subscription 1",
    "state": "Enabled",
    "tenantId": "5b234a67-1ad7-4021-9056-328b95f63810",
    "user": {
      "name": "scott@mcclung.com",
      "type": "user"
    }
  },
  {
    "cloudName": "AzureCloud",
    "homeTenantId": "5b234a67-1ad7-4021-9056-328b95f63810",
    "id": "aa79e39f-b196-4705-9073-85746c273a97",
    "isDefault": false,
    "managedByTenants": [],
    "name": "Azure Testing",
    "state": "Disabled",
    "tenantId": "5b234a67-1ad7-4021-9056-328b95f63810",
    "user": {
      "name": "scott@mcclung.com",
      "type": "user"
    }
  }
]
```

That "Azure Testing" AAD was something I created for learning, and I haven't yet figured out how to get rid of it entirely, but it's not the problem.   The "wrong" account has no subscription at all.

**Comment by hholst80** [comment]
I just moved my `~/.azure` directory out of the way and ran the command again. Now it worked.

**Comment by alan994** [comment]
Hi guys, it seems I have same problem :(
I installed Azure CLI on my Ubuntu box. Standard installation from docs. 

When I tried to run `az login` I got the error from the first post. After running `az account clear` and then `az login` it works ok and I was able to set my account subscription `az account set --subscription <subscription_name>. Next step for me is to run `az acr login -n <container_registry_name>` this command should work without problem but it asks me my credentials and after I enter my credentials I got folowing error. 

```
sudo az acr login -n <my_container_registry_name>                                                                                                  ─╯
Unable to get AAD authorization tokens with message: Please run 'az login' to setup account.
Unable to get admin user credentials with message: Please run 'az login' to setup account.
Username: <my_email>
Password: 
Error response from daemon: Get "https://<my_registry_name>.azurecr.io/v2/": unauthorized: Application not registered with AAD.
Login failed.
```

Any help with this?

**Comment by jiasli** [comment]
@alan994, your issue regarding `az acr login` is a separate one. I have split it into https://github.com/Azure/azure-cli/issues/21083 and asked ACR service team to take a look.

**Comment by mgrazebrook** [comment]
Same problem. I set up a starter Azure account with $200 credit. All fine for a few days. Then I had weird login problems. Then I could login but could not see my subscription. Support reported " I see that it is recognized as a “Personal Account” and as well as “Work or School Account”" and gave instructions starting "Open Edge browser | InPrivate mode" which I followed (but using Chrome/Incognito). I can now only see my account in Incognito mode.

Using az login from the command line (I'm doing a CLI training module), I get a message:
```
Found multiple accounts with the same username 'michael@grazebrook.com': 
(snip 2 records - I imagine it's insecure to publish the values!)
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168
```

**Comment by blue928** [comment]
***This is URGENT***.  Please advise steps to regain access to my professional work and workspace.

I have had a Microsoft account for personal use / xbox for years. I signed up for DevOps and other technology services with the same email that I accessed through portal.azure.com. Created a significant amount of work over the past several weeks. Today, I was asked to change my password. When I did, I got several errors which led me here. I finally clicked through enough buttons that when I got to portal.azure.com, where I previously saw my technology resources (mysql server, kubernetes, etc),  I now see my PERSONAL info that discusses my xbox subscription, family account stuff, etc.

`az login` or `az login -t <my professional tenant id>` all lead to login loops.

I've lost access to ALL my professional work and have no idea how to get back. I've deleted the .azure folder; I've ran `az account clear`; nothing works. Please advise!

This is the error I keep getting:

```
Found multiple accounts with the same username 'myname@myemail.com': [{'home_account_id': '00000000-0000-0000-d399-d6d1d4a4ca7a.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'myname@myemail.com', 'authority_type': 'MSSTS', 'local_account_id': '618057b1-4c21-4e36-b684-03135cbcbed3', 'realm': '0457c22a-03ab-4ceb-8e26-391ef8087bc2'}, {'home_account_id': 'b9d64a87-8413-4861-8cce-40ad8f530962.30ff9873-366e-4653-8ff0-63f6d2ebf293', 'environment': 'login.microsoftonline.com', 'username': 'myname@myemail.com, 'authority_type': 'MSSTS', 'local_account_id': 'GAmuQnC8tFfz0fRU6JAQy7FjmpWT4l5zIQdT1tVPhRE', 'realm': '0457c22a-03ab-4ceb-8e26-391ef8087bc2'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168
```

**Comment by jiasli** [comment]
@blue928, 

> I finally clicked through enough buttons that when I got to portal.azure.com, where I previously saw my technology resources (mysql server, kubernetes, etc), I now see my PERSONAL info that discusses my xbox subscription, family account stuff, etc.

Based on my understanding, you cannot sign in to Azure Portal either, right? For that case, could you clear the browser cache or use InPrivate or Incognito mode?

> I've deleted the .azure folder; I've ran `az account clear`; nothing works.

After clearing the browser cache, run `az account clear` and `az login`, but only select the account you use to access Azure resources, and see if the error is gone.

**Comment by robch** [comment]
I ran into this same issue. How can I fix it?

**Comment by yonzhan** [comment]
Please adopt the approach suggested in the last comment.

**Comment by AdamDudley** [comment]
Hope this helps.

Steps to reproduce

1. `az login`
2. choose your 'work' account for the email address
3. `az account set -s "subscription"`
4. choose the different account (personal) for the same email address as 2.
5. get the error

fix is

```
az account clear
az login
```

and use the same 'account' for the email address each time... user error, no bug here.

**Comment by c240amg** [comment]
The above works for me.
Issue in my case was caused by a tenancy migration

**Comment by vnomus** [comment]
Same issue here,  but above solution does not work for me. If, when I'm presented with the two matching accounts to chose from during login (after clearing account):
1. Work account - I get a "No subscription found error".
2. Personal account - log in and I can see my MSDN subscription. However, I can no longer access the artifacts repository created by that subscription.

**Comment by wmeints** [comment]
Same issue here! I deleted the token cache and was able to login as normal. I have one personal account and a work account with the same name. Can't rename them or else my MVP profile becomes inaccessible :(

**Comment by Mastansourcefuse** [comment]
Found multiple accounts with the same username 'sk.mastan@sourcefuse.com': [{'home_account_id': '4cc03aea-8a4c-4092-a20a-9b3653079ed2.a86a5cb3-752e-49a9-82e6-dd80ee4bf2c2', 'environment': 'login.microsoftonline.com', 'username': 'sk.mastan@sourcefuse.com', 'authority_type': 'MSSTS', 'local_account_id': '4cc03aea-8a4c-4092-a20a-9b3653079ed2', 'realm': 'organizations'}, {'home_account_id': '263e78d2-19cf-48dc-a8f0-6f478b71fcbc.ddb21e3d-29dd-4280-a4dd-d05a97bc050c', 'environment': 'login.microsoftonline.com', 'username': 'sk.mastan@sourcefuse.com', 'authority_type': 'MSSTS', 'local_account_id': '263e78d2-19cf-48dc-a8f0-6f478b71fcbc', 'realm': 'organizations'}]

**Comment by Revlig** [comment]
> No problem. I deleted the json files in the .azure directory :)

same here :D

**Comment by Rakigh** [comment]
> Found multiple accounts with the same username '[sk.mastan@sourcefuse.com](mailto:sk.mastan@sourcefuse.com)': [{'home_account_id': '4cc03aea-8a4c-4092-a20a-9b3653079ed2.a86a5cb3-752e-49a9-82e6-dd80ee4bf2c2', 'environment': 'login.microsoftonline.com', 'username': '[sk.mastan@sourcefuse.com](mailto:sk.mastan@sourcefuse.com)', 'authority_type': 'MSSTS', 'local_account_id': '4cc03aea-8a4c-4092-a20a-9b3653079ed2', 'realm': 'organizations'}, {'home_account_id': '263e78d2-19cf-48dc-a8f0-6f478b71fcbc.ddb21e3d-29dd-4280-a4dd-d05a97bc050c', 'environment': 'login.microsoftonline.com', 'username': '[sk.mastan@sourcefuse.com](mailto:sk.mastan@sourcefuse.com)', 'authority_type': 'MSSTS', 'local_account_id': '263e78d2-19cf-48dc-a8f0-6f478b71fcbc', 'realm': 'organizations'}]

Did you get the solution for this

**Comment by D11bexell** [comment]
Hello! Im this error too! I set my default browser to Edge and deleted all cache & cookies. Still the same error after login attempt. Any help appreciated
![image](https://user-images.githubusercontent.com/52918186/207651181-d4f31490-acf1-4152-aa91-44555d1549c0.png)

**Comment by Rakigh** [comment]
Even i am facing the same problem.

**Comment by jrthib** [comment]
Same issue here as well

**Comment by joestringer** [comment]
I started having this problem today.

**Comment by Aki-Siponen** [comment]
Hi,
As requested I'll report the issue:
```
Found multiple accounts with the same username 'aki@siponen.fi': [{'home_account_id': 'ad1fb935-c50f-4b07-8ebe-bc82f2efedd0.58a40091-38c3-4621-884a-026b6b95bd1e', 'environment': 'login.microsoftonline.com', 'username': 'aki@siponen.fi', 'authority_type': 'MSSTS', 'local_account_id': 'ad1fb935-c50f-4b07-8ebe-bc82f2efedd0', 'realm': 'organizations'}, {'home_account_id': '00000000-0000-0000-726c-8ac3fed1fb58.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'aki@siponen.fi', 'authority_type': 'MSSTS', 'local_account_id': 'DUPZnRKweTkgq6RDJjgL-c16xu7aNUG8dUQQd0aFxrg', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

```
when logging in to Azure CLI in VS Code PowerShell.
**az account clear solved the issue and I could log in. Thank you for the advice!**

One of those accounts is Microsoft account and the other is work/school account for my private domain.

I see that there is some discussion about how this is possible. There is at least one possible workflow to create the situation.
I don't know if this is still possible. I think it is:

1. set up a domain, let's say mydomain.com
2. use Google or some other email provider for the domain
3. create email myname@mydomain.com
4. create a Microsoft account using the email myname@mydomain.com
5. purchase Microsoft 365, you get domain mydomain.onmicrosoft.com
6. transfer your domain (mydomain.com) and email to Microsoft 365
7. now you have a work email myname@mydomain.com

I did this in 2008 - 2011.

**Comment by jmleddy** [comment]
Turning off token encryption and removing the token cache file resolved it for me as well. On mac `rm ~/.azure/msal_token_cache.json`

**Comment by rayluo** [comment]
> Turning off token encryption and removing the token cache file resolved it for me as well. On mac `rm ~/.azure/msal_token_cache.json`

While you are removing the token cache file anyway, the token encryption is irrelevant. Besides, I believe Azure CLI does not yet encrypt tokens on mac.

**Comment by robch** [comment]
Reporting as requested:

![image](https://user-images.githubusercontent.com/1608910/232180015-716e6f57-7658-4312-b34c-84c65124ad77.png)

**Comment by JamesMHerrington** [comment]
Reporting as directed:

Found multiple accounts with the same username 'JamesM.Herrington@am.jll.com': [{'home_account_id': '1d2573cc-12cf-45ab-b3c3-166139d7f7a9.bfef2b06-d256-4f8e-bd03-8d3687987063', 'environment': 'login.microsoftonline.com', 'username': 'JamesM.Herrington@am.jll.com', 'authority_type': 'MSSTS', 'local_account_id': '1d2573cc-12cf-45ab-b3c3-166139d7f7a9', 'realm': 'organizations'}, {'home_account_id': '00000000-0000-0000-de1b-4e5a5c2478ff.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'JamesM.Herrington@am.jll.com', 'authority_type': 'MSSTS', 'local_account_id': '00000000-0000-0000-de1b-4e5a5c2478ff.9188040d-6c67-4c5b-b112-36a304b66dad.f8cdef31-a31e-4b4a-93e4-5f571e91255a', 'realm': 'organizations'}

**Comment by bhollowa** [comment]
Reporting as requested:

Found multiple accounts with the same username 'benjamin.holloway@magnet.cl': [{'home_account_id': 'c4e998be-d05e-43d3-9fa5-34c669965194.3f6e4746-ca7a-4fd2-8994-5ad7b4df62b0', 'environment': 'login.microsoftonline.com', 'username': 'benjamin.holloway@magnet.cl', 'authority_type': 'MSSTS', 'local_account_id': 'c4e998be-d05e-43d3-9fa5-34c669965194', 'realm': 'organizations'}, {'home_account_id': '00000000-0000-0000-96ca-c1b80e94862f.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'benjamin.holloway@magnet.cl', 'authority_type': 'MSSTS', 'local_account_id': '1e819715-e63c-46b1-b778-272b5fbc4f19', 'realm': 'bf5b97bd-7c69-4424-9dd3-72af5a8ac4af'}]

**Comment by ehershey** [comment]
I got this error too. 
```
Found multiple accounts with the same username 'ernie.hershey@xxx': [{'home_account_id': 'xxx', 'environment': 'login.microsoftonline.com', 'username': 'ernie.hershey@xxx', 'authority_type': 'MSSTS', 'local_account_id': 'xxx', 'realm': 'organizations'}, {'home_account_id': 'xxx', 'environment': 'login.microsoftonline.com', 'username': 'ernie.hershey@xxx', 'authority_type': 'MSSTS', 'local_account_id': 'xxx', 'realm': 'organizations'}]
```

**Comment by craigmj** [comment]
Reporting as directed: Found multiple accounts with the same username 'craig@lateral.co.za': [{'home_account_id': '00000000-0000-0000-226c-4e0a43c67914.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'craig@lateral.co.za', 'authority_type': 'MSSTS', 'local_account_id': 'kISun9DEDcR3pNu7H8kd4CS8PbJ0XXkmW2yMvziPqiQ', 'realm': 'organizations'}, {'home_account_id': '3eec8dab-c90d-4871-a539-bf53a88f25ad.589bcab5-cea5-4eaa-ba34-9e58b9d21b40', 'environment': 'login.microsoftonline.com', 'username': 'craig@lateral.co.za', 'authority_type': 'MSSTS', 'local_account_id': '3eec8dab-c90d-4871-a539-bf53a88f25ad', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

I need this resolved asap. I've got a client who won't have any service until I can access the cloud again...

**Comment by pwae** [comment]
I had this issue too. As a summary, we previously used Personal MS accounts (with our work email) to sign in.
We then migrated from that to Azure AD, and migrate the devops tenant.

I suspect this is related as to why the errors occur.

**Comment by HenriKoelewijn** [comment]
I had this issue too. 
Used 
az config set core.token_encryption=false
az login

and followed the directions that were printed in the browser.
That worked.

**Comment by sarveswarreddy** [comment]
You can cleanup one of the entries of the same account in the json file under your home directory ".azure/azureProfile.json"
It solved the issue for me..

**Comment by aaltotsky** [comment]
I experienced the same issue. I do have 2 accounts with the same email name. One "**Personal**" and another "**Work**". The personal account has access to the correct subscription. It worked fine before until today. 
Here is the information about the `az` version:
```
C:\Users\master>az --version
azure-cli                         2.49.0

core                              2.49.0
telemetry                          1.0.8

Extensions:
containerapp                      0.3.33

Dependencies:
msal                              1.20.0
azure-mgmt-resource               22.0.0

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\master\.azure\cliextensions'

Python (Windows) 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:05:00) [MSC v.1929 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
```

**Comment by aaltotsky** [comment]
I found a solution.

1. Uninstall AZ msi See: https://learn.microsoft.com/en-us/cli/azure/install-azure-cli-windows?view=azure-cli-latest&tabs=powershell#uninstall 
2. Run PowerShell command:  `Install-Module -Name Az -Repository PSGallery -Force -AllowClobber`
3. Install the latest MSI: https://aka.ms/installazurecliwindows
4. Use `Connect-AzAccount -TenantId <TenantID>`
5. Continue to use the `az` command (without using `az login`). 

You may try steps 4-5 without steps 1 to 3.

**Comment by Ryangr0** [comment]
> I found a solution.
> 
> Use `Connect-AzAccount -TenantId <TenantID>`

I think we're experiencing the same exact issue. I don't think that using powershell instead of the cli is a proper solution to this problem. I wonder if there's a way to merge these accounts? This isn't the only area in which I'm experiencing strange behavior within the azure ecosystem, and I'm getting more and more convinced me having both a personal and a work account on the same email is the problem.

**Comment by aaltotsky** [comment]
Here is another workaround: 

1. Sign in to the Azure Portal (https://portal.azure.com/)
2. Click on the Cloud Shell (Next to the search box).
3. Use `az login` there.

**Comment by dodalovicgran** [comment]
```
❯ az login
A web browser has been opened at https://login.microsoftonline.com/organizations/oauth2/v2.0/authorize. Please continue the login in the web browser. If no web browser is available or if the web browser fails to open, use device code flow with `az login --use-device-code`.
Opening in existing browser session.
Found multiple accounts with the same username 'dusan.odalovic@gransoftware.de': [{'home_account_id': '8a5134bc-65f5-4de8-9611-9ed50f821da8.3e9ceed8-1dd5-422c-8d37-b26422dce745', 'environment': 'login.microsoftonline.com', 'username': 'dusan.odalovic@gransoftware.de', 'authority_type': 'MSSTS', 'local_account_id': '8a5134bc-65f5-4de8-9611-9ed50f821da8', 'realm': '3e9ceed8-1dd5-422c-8d37-b26422dce745'}, {'home_account_id': '00000000-0000-0000-cdb6-0b41b24413e2.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'dusan.odalovic@gransoftware.de', 'authority_type': 'MSSTS', 'local_account_id': '331c08ab-44b9-449c-b2d9-4e0e18d347ef', 'realm': '3e9ceed8-1dd5-422c-8d37-b26422dce745'}]
```

**Comment by craigmj** [comment]
I still have this problem:
```Found multiple accounts with the same username 'craig@lateral.co.za': [{'home_account_id': '00000000-0000-0000-226c-4e0a43c67914.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'craig@lateral.co.za', 'authority_type': 'MSSTS', 'local_account_id': 'kISun9DEDcR3pNu7H8kd4CS8PbJ0XXkmW2yMvziPqiQ', 'realm': 'organizations'}, {'home_account_id': '3eec8dab-c90d-4871-a539-bf53a88f25ad.589bcab5-cea5-4eaa-ba34-9e58b9d21b40', 'environment': 'login.microsoftonline.com', 'username': 'craig@lateral.co.za', 'authority_type': 'MSSTS', 'local_account_id': '3eec8dab-c90d-4871-a539-bf53a88f25ad', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168
```
I'm struggling to understand how you can be running a cloud service where users get logged out of their accounts. We are starting to migrate all our infrastructure off your cloud, to a more 'accessile' one...

**Comment by craigmj** [comment]
Here's the error - github removed it for some reason:

Found multiple accounts with the same username 'craig@lateral.co.za': [{'home_account_id': '00000000-0000-0000-226c-4e0a43c67914.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'craig@lateral.co.za', 'authority_type': 'MSSTS', 'local_account_id': 'kISun9DEDcR3pNu7H8kd4CS8PbJ0XXkmW2yMvziPqiQ', 'realm': 'organizations'}, {'home_account_id': '3eec8dab-c90d-4871-a539-bf53a88f25ad.589bcab5-cea5-4eaa-ba34-9e58b9d21b40', 'environment': 'login.microsoftonline.com', 'username': 'craig@lateral.co.za', 'authority_type': 'MSSTS', 'local_account_id': '3eec8dab-c90d-4871-a539-bf53a88f25ad', 'realm': 'organizations'}]

**Comment by MuhammadSalmanGalileo** [comment]
I have same problem here is the error 
Found multiple accounts with the same username 'm.salman@efishery.com': [{'home_account_id': 'a1e2efba-95f3-4703-befb-0ebe01d062d2.aefbf958-7519-4710-990a-d2312cfe4fc2', 'environment': 'login.microsoftonline.com', 'username': 'm.salman@efishery.com', 'authority_type': 'MSSTS', 'local_account_id': 'a1e2efba-95f3-4703-befb-0ebe01d062d2', 'realm': 'organizations'}, {'home_account_id': '00000000-0000-0000-63eb-d2114b254395.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'm.salman@efishery.com', 'authority_type': 'MSSTS', 'local_account_id': 'OORfRCtkTyOBLt4FdLmekmPhVtg7s79fhOvReBnP0Dk', 'realm': 'organizations'}]

**Comment by BrunoJuchli** [comment]
>ERROR: Found multiple accounts with the same username 'bruno.juchli@smart-me.com': [{'home_account_id': 'd32db2dd-4414-4cdd-917e-c2f1cb3b8e85.5c9a4ff6-00e2-4775-9e18-cc5e6745a1f9', 'environment': 'login.microsoftonline.com', 'username': 'bruno.juchli@smart-me.com', 'authority_type': 'MSSTS', 'local_account_id': 'd32db2dd-4414-4cdd-917e-c2f1cb3b8e85', 'realm': 'organizations'}, {'home_account_id': '01948d69-2651-41cd-9815-d75892fb52a0.67d2b627-4607-4f1d-b430-2760715a2ae2', 'environment': 'login.microsoftonline.com', 'username': 'bruno.juchli@smart-me.com', 'authority_type': 'MSSTS', 'local_account_id': '01948d69-2651-41cd-9815-d75892fb52a0', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

**Comment by SusserBank** [comment]
Same error:
╷
│ Error: obtaining Authorization Token from the Azure CLI: parsing json result from the Azure CLI: waiting for the Azure CLI: exit status 1: ERROR: Found multiple accounts with the same username 'domadminswinters@susser.bank': [{'home_account_id': '012156d7-6e1b-4ef1-b9d2-41a9ec136447.518917f1-fecb-440c-b871-99406caf2528', 'environment': 'login.microsoftonline.com', 'username': 'domadminswinters@susser.bank', 'authority_type': 'MSSTS', 'local_account_id': '012156d7-6e1b-4ef1-b9d2-41a9ec136447', 'realm': 'organizations'}, {'home_account_id': 'ef6e89f8-6656-442e-8dd7-676a2651b8d5.518917f1-fecb-440c-b871-99406caf2528', 'environment': 'login.microsoftonline.com', 'username': 'domadminswinters@susser.bank', 'authority_type': 'MSSTS', 'local_account_id': 'ef6e89f8-6656-442e-8dd7-676a2651b8d5', 'realm': 'organizations'}]
│ Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

az account clear
az login 
resolved the issue

**Comment by bburdette** [comment]
Just got directed here by `az login`, "Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168".  

The error was "Found multiple accounts with the same username", followed by specific account information I won't post here.  

My situation is I was sent an azure login link by an admin and I've been using that to manage things in azure.  That azure account hasn't been associated with a microsoft account, though it does have an email associated with it, "emailA".  I created a microsoft account at some point with a different email address ("emailB"), but somehow it got associated with the emailA even though that was not listed on my user information profile.  Not really sure of the sequence of events there, its been over a year I think.

Anyway, recently I was trying to `az login` and it wanted a microsoft account.  I tried creating one with emailA, but it said that was already associated with a microsoft account, even though I couldn't log in with emailA.  I went through the process of creating a new account with emailA, but it would get to a "Create Account" dialog and hang.  

So I tried deleting the emailB microsoft account, but still could not create a microsoft account under emailA.  I did not wait the required 60 days for full deletion.  

So I logged in with emailB (reviving that microsoft account) and changed the email on that account to emailA, then removed emailB from it.

**Comment by SusserBank** [comment]
Thanks. It’s a Windows server, so this worked for me:

az account clear
az login


Stormy Winters
System Administrator

Susser Bank
3030 Matlock Road | Arlington, Texas 76015
Direct: 817.987.2178

***@***.***

**Comment by bburdette** [comment]
> On Ubuntu **22.04** `rm ~/.azure/*.json` will solve the issue.

This fixed my `az login` issue.  Along the way I also changed the email on my non-work microsoft account so it wouldn't conflict with the work azure account with the same email.

**Comment by michaelfeil** [comment]
> On Ubuntu 22.04 rm ~/.azure/*.json will solve the issue.


This helped also to resolve this:
```
(evalenv) michi@michi:~/grdt$ az login
A web browser has been opened at https://login.microsoftonline.com/organizations/oauth2/v2.0/authorize. Please continue the login in the web browser. If no web browser is available or if the web browser fails to open, use device code flow with `az login --use-device-code`.
Found multiple accounts with the same username 'michael@me.ai': [{'home_account_id': '00000000-0000-0000-db93-547fxxxxxxxxxxxxxxx-6c67-4c5b-b112-xxxxx', 'environment': 'login.microsoftonline.com', 'username': 'michael@me.ai', 'authority_type': 'MSSTS', 'local_account_id': 'ySpdjOrR8rByUI2sxIOGaroF9vXp-GlNvYmKGTvbiwQ', 'realm': 'organizations'}, {'home_account_id': '0f5e035b-fe69-4fbd-aced-be134d871024.a4d0a6ec-5402-xxxxxxxx-xxxxxx', 'environment': 'login.microsoftonline.com', 'username': 'michael@me.ai', 'authority_type': 'MSSTS', 'local_account_id': 'xxxxxxxx-fe69-4fbd-aced-xxxxxxx', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168
```

**Comment by MNO098** [comment]
I followed all possible option mentioned here in my windows machine but it dint work for me,

I tried,
az account clear
az login

once I login with my user account , I am getting below error,

033ET744> az login
Please select the account you want to log in with.
<urllib3.connection.HTTPSConnection object at 0x07057530>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed

@jiasli kindly help to resolve this issue

**Comment by jziggas** [comment]
I'm not entirely sure but I believe I resolved this issue with the following steps (because any of the individual suggestions above not work):

- Going to portal.azure.com and logging out to remove any browser session
- Running `az account clear` in the terminal
- Running `rm ~/.azure/*.json`
- Running ` az login --tenant TENANT_ID --use-device-code`

**Comment by jonas-lomholdt** [comment]
Thanks! I did a `rm ~/.azure/msal_token_cache.json` and that solved it for me.

**Comment by rayluo** [comment]
Food for thought, @jiasli.

We have been collecting the user reports for this issue for a while now. The "multiple accounts with same username" symptoms covered a wide spectrum.

* [Some cases contained exact same account](https://github.com/Azure/azure-cli/issues/20168#issuecomment-960164017) (and now those cases are bypassed by MSAL automatically grouping those same accounts).
* [Some cases contained an MSA account and an AAD account](https://github.com/Azure/azure-cli/issues/20168#issuecomment-960386197).
* [Some cases contained two AAD accounts in different tenants](https://github.com/Azure/azure-cli/issues/20168#issuecomment-17640563660), some in the [same tenant](https://github.com/Azure/azure-cli/issues/20168#issuecomment-1830403728).
* The [latest report above](https://github.com/Azure/azure-cli/issues/20168#issuecomment-2257765873) even gave us more hint by the recent `account_source` info so we know in that particular case, at least one of the accounts was signed in via device code flow.

The problem is that we still don't really know why we ended up with such a state in the first place, despite [sporadic repro report(s)](https://github.com/Azure/azure-cli/issues/20168#issuecomment-1835013169). We just knew that username is not unique, and Entra ID somehow has no problem allowing different accounts with same username to sign in and knows who-is-who.

I suppose on the client side, and Azure CLI in particular, we could either:

1. Add an experimental check after each successful `az login ...` to proactively check `get_accounts(username=current_username_obtained_via_ID_token)` and fail early with same error message. That way, when everything else being equal, at least we will have better idea of how the 2nd account would have been established via an "az login".
2. Alternatively, MSAL could add a `last_modification_time` field into the accounts so that Azure CLI may use it to automatically choose the newer account to proceed, optionally automatically removing the old account. This essentially automates the "az account clear; az login ..." workaround.

**Comment by mjmarcum** [comment]
I have the same problem and was directed here: 

 'michael@oahutechnology.com': 
 [{'home_account_id': '00000000-0000-0000-3eac-1861bc270a84.9188040d-6c67-4c5b-b112-36a304b66dad', 
   'environment': 'login.microsoftonline.com',
   'username': 'michael@oahutechnology.com', 
   'account_source': 'authorization_code', 
   'authority_type': 'MSSTS', 
   'local_account_id': 'LKp369dl9OE6DHMonYlFiiDPH4Tii9sUT9H_j6mw_bg', 
   'realm': 'organizations'}, 
  {'home_account_id': 'b564f479-80c5-42c5-bdd1-90a33fa7dd3f.52e9439d-02fc-48c8-b0ad-318955628465', 
   'environment': 'login.microsoftonline.com', 
   'username': 'michael@oahutechnology.com', 
   'account_source': 'broker', 
   'authority_type': 'MSSTS', 
   'local_account_id': 'b564f479-80c5-42c5-bdd1-90a33fa7dd3f', 
   'realm': 'organizations'}]

**Comment by mjmarcum** [comment]
Same issue not sure how to merge these accounts

Found multiple accounts with the same username 'michael@oahutechnology.com': [{'home_account_id': '00000000-0000-0000-3eac-1861bc270a84.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'michael@oahutechnology.com', 'account_source': 'authorization_code', 'authority_type': 'MSSTS', 'local_account_id': 'LKp369dl9OE6DHMonYlFiiDPH4Tii9sUT9H_j6mw_bg', 'realm': 'organizations'}, {'home_account_id': 'b564f479-80c5-42c5-bdd1-90a33fa7dd3f.52e9439d-02fc-48c8-b0ad-318955628465', 'environment': 'login.microsoftonline.com', 'username': 'michael@oahutechnology.com', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': 'b564f479-80c5-42c5-bdd1-90a33fa7dd3f', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

**Comment by asouzajunior** [comment]
Same problem here . Tried to use az login, but get this message. My e-mail was omitted below.

% az login  
A web browser has been opened at https://login.microsoftonline.com/organizations/oauth2/v2.0/authorize. Please continue the login in the web browser. If no web browser is available or if the web browser fails to open, use device code flow with `az login --use-device-code`.

Retrieving tenants and subscriptions for the selection...
Found multiple accounts with the same username 'axxxxxza@xxxxx.com.br': [{'home_account_id': '16223eeb-871b-47ed-a948-cd5ba9241cc4.995266c5-aa22-415a-afc2-2cc026ce99f9', 'environment': 'login.microsoftonline.com', 'username': 'axxxxxza@xxxxx.com.br', 'account_source': 'authorization_code', 'authority_type': 'MSSTS', 'local_account_id': '16223eeb-871b-47ed-a948-cd5ba9241cc4', 'realm': 'organizations'}, {'home_account_id': '00000000-0000-0000-0c83-47e3c67e0948.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'axxxxxza@xxxxx.com.br', 'account_source': 'authorization_code', 'authority_type': 'MSSTS', 'local_account_id': 'HbyDaX5XjgMbvbQ0VVnF_58o-Ch_9DaeSQBatAgrRZE', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

**Comment by zyptai** [comment]
Same issue.  My issue is my user thinks i have a personal and a microsoft account in the same user ID.

ERROR: Found multiple accounts with the same username 'tixxxxx@i.com': [{'home_account_id': '747e0494-612a-49d2-a9f0-ac512e4340ae.f20f5397-6421-44f6-80aa-1dbcc827e24d', 'environment': 'login.microsoftonline.com', 'username': 'tixxxxx@i.com':, 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '747e0494-612a-49d2-a9f0-ac512e4340ae', 'realm': 'f20f5397-6421-44f6-80aa-1dbcc827e24d'}, {'home_account_id': '00000000-0000-0000-edbe-a05df11c0122.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': ''tixxxxx@i.com':, 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '00000000-0000-0000-edbe-a05df11c0122.9188040d-6c67-4c5b-b112-36a304b66dad.f8cdef31-a31e-4b4a-93e4-5f571e91255a', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

**Comment by dmwozniak-grg** [comment]
Same issue.

azure.core.exceptions.ClientAuthenticationError: ERROR: Found multiple accounts with the same username 'contact@dmwozniak.com': [{'home_account_id': '6793ad6a-e2b9-4b04-9ae4-001e9a505e49.56d2060f-2bdb-4206-80a1-4a09f5f7e9ed', 'environment': 'login.microsoftonline.com', 'username': 'contact@dmwozniak.com', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '6793ad6a-e2b9-4b04-9ae4-001e9a505e49', 'realm': '56d2060f-2bdb-4206-80a1-4a09f5f7e9ed'}, {'home_account_id': '00000000-0000-0000-4d7d-e1fd64222eed.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'contact@dmwozniak.com', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '00000000-0000-0000-4d7d-e1fd64222eed.9188040d-6c67-4c5b-b112-36a304b66dad.f8cdef31-a31e-4b4a-93e4-5f571e91255a', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

**Comment by Rik-de-Kort** [comment]
Same here: I do have a personal and a work microsoft account, and would very much like to be rid of the personal one! It's not used for anything.

```powershell
C:\src\backend> az login
Select the account you want to log in with. For more information on login with Azure CLI, see https://go.microsoft.com/fwlink/?linkid=2271136

Retrieving tenants and subscriptions for the selection...
Found multiple accounts with the same username 'rik.de.kort@pggm.nl': [{'home_account_id': '00000000-0000-0000-befc-f04dd7b64129.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'rik.de.kort@pggm.nl', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': 'a57d4441-a938-46cc-9ac8-2aabcbeb22a5', 'realm': 'fa9e26e0-015d-45c0-9d54-2af246e6ddf2'}, {'home_account_id': 'eeb928d6-4e33-48fb-8eed-74de067e512a.9e3ee5fe-3a99-45db-ac6e-691e86febef3', 'environment': 'login.microsoftonline.com', 'username': 'rik.de.kort@pggm.nl', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '1d2b51c3-21e9-41d6-bb68-edf52d0d93bd', 'realm': 'bf8ba62d-740b-46e9-a7da-cfe794c7da80'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168
```

**Comment by maxgolov** [comment]
I have a repro of this after my account got deleted in Entra. Then recreated again with the same username / login.

**Comment by spencercg** [comment]
> ## Solution
> 
> To get unblocked, please clear Azure CLI's accounts and log in again with only one account (Microsoft Account or AAD Account):
> 
> ```
> az account clear
> az login
> ```
> 
> Later if you want to logging in again, please make sure to select the same account, but not the other one.

This solved the problem for me, thanks so much!

**Comment by euangordon** [comment]
Same issue here, tried above solutions can't get it to work. Authenticator app is popping up on my phone but no code is visible to enter

**********************************************************************
** Visual Studio 2022 Developer PowerShell v17.12.5
** Copyright (c) 2022 Microsoft Corporation
**********************************************************************
PS C:\Users\Euan Gordon\source\repos\CompassCore-Functions> az account clear
PS C:\Users\Euan Gordon\source\repos\CompassCore-Functions> az login
Select the account you want to log in with. For more information on login with Azure CLI, see https://go.microsoft.com/fwlink/?linkid=2271136

Retrieving tenants and subscriptions for the selection...
Authentication failed against tenant 1cbf1c7a-1cc6-4f7d-90f3-e76b01ea1a10 'DSNM Ltd': (pii). Status: Response_Status.Status_InteractionRequired, Error code: 3399614476, Tag: 557973645
The following tenants don't contain accessible subscriptions. Use `az login --allow-no-subscriptions` to have tenant level access.
94f6ed9d-c518-40e6-9c96-8eedc4b79ad3 'nautoshark.com'
If you need to access subscriptions in the following tenants, please use `az login --tenant TENANT_ID`.
1cbf1c7a-1cc6-4f7d-90f3-e76b01ea1a10 'DSNM Ltd'
No subscriptions found for euan.gordon@nautoshark.com.
PS C:\Users\Euan Gordon\source\repos\CompassCore-Functions> az login --tenant 1cbf1c7a-1cc6-4f7d-90f3-e76b01ea1a10
Select the account you want to log in with. For more information on login with Azure CLI, see https://go.microsoft.com/fwlink/?linkid=2271136

**Comment by gerwim** [comment]
Can't login either:

```
➜  ~ az login
A web browser has been opened at https://login.microsoftonline.com/organizations/oauth2/v2.0/authorize. Please continue the login in the web browser. If no web browser is available or if the web browser fails to open, use device code flow with `az login --use-device-code`.

Retrieving tenants and subscriptions for the selection...
Found multiple accounts with the same username 'xxxxxxx': [{'home_account_id': '8fd750ec-d7d0-4073-a346-b12948ee14a3.fb028871-408d-4624-9101-996f397d2b95', 'environment': 'login.microsoftonline.com', 'username': 'xxxxxxx', 'account_source': 'authorization_code', 'authority_type': 'MSSTS', 'local_account_id': '8fd750ec-d7d0-4073-a346-b12948ee14a3', 'realm': 'organizations'}, {'home_account_id': '00000000-0000-0000-e5e7-5d09b587c631.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'xxxxxxx', 'account_source': 'authorization_code', 'authority_type': 'MSSTS', 'local_account_id': 'jTAqNAsk7za_BXRcZsvRdbyT4Rq5srXSyHVfaS0TRI0', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168
```

**Comment by keystroke** [comment]
Running into this in edge / private cloud contexts. Our STS can support "Entra" style issuers as well as "ADFS" style issuers, and when tooling mixes these, we get errors with the token cache finding the same user twice:

Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

ERROR: az_command_data_logger: Found multiple accounts with the same username 'admin@demo.asz': [{'home_account_id': 
'[redacted-guid-1].None', 'environment': '[redacted-name]', 'username': 
'[redacted-username]', 'account_source': 'password', 'authority_type': 'MSSTS', 'local_account_id': 
'[redacted-guid-1]', 'realm': '[redacted-guid-2]'}, {'home_account_id': 
'KZjAQs8ePotMJg3tIQDvqg', 'environment': '[redacted-name]', 'username': '[redacted-username]', 
'account_source': 'password', 'authority_type': 'ADFS', 'local_account_id': '[redacted-guid-1]', 
'realm': 'adfs'}]

Not sure why the ".None" suffix on the home account id in the first example.

**Comment by euangordon** [comment]
Just as a helper for anyone else with these issues, your life will be 20x better if you manage to have only one account with the same email address. 

It took a little time, but I managed to get all subscriptions linked to one of my accounts, the other account was an Owner of another subscription / 365 Org, so I couldn't delete it, but I could change the email address. By changing that, I now only have one account linked to my email and everything works great. The amount of MFA I have to do while logging in is about 1% of what it used to be, honestly worth the initial pain.

Long Story Short, having multiple accounts really doesn't work.

**Comment by keystroke** [comment]
> Looks like this error comes from here in the Azure CLI: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/auth/msal_credentials.py#L36
> 
> accounts = self._msal_app.get_accounts(username)
> 
> # Usernames are usually unique. We are collecting corner cases to better understand its behavior.
> if len(accounts) > 1:
>     raise CLIError(f"Found multiple accounts with the same username '{username}': {accounts}\n"
>                    "Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168")
> The code is expecting that MSAL Python cache will have only a single username account in token cache.
> 
> This is the call in MSAL Python: https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/dev/msal/application.py#L1216
> 
> def get_accounts(self, username=None):
>         """Get a list of accounts which previously signed in, i.e. exists in cache.
> 
>         An account can later be used in :func:`~acquire_token_silent`
>         to find its tokens.
> 
>         :param username:
>             Filter accounts with this username only. Case insensitive.
>         :return: A list of account objects.
>             Each account is a dict. For now, we only document its "username" field.
>             Your app can choose to display those information to end user,
>             and allow user to choose one of his/her accounts to proceed.
>         """
>         accounts = self._find_msal_accounts(environment=self.authority.instance)
>         if not accounts:  # Now try other aliases of this authority instance
>             for alias in self._get_authority_aliases(self.authority.instance):
>                 accounts = self._find_msal_accounts(environment=alias)
>                 if accounts:
>                     break
>         if username:
>             # Federated account["username"] from AAD could contain mixed case
>             lowercase_username = username.lower()
>             accounts = [a for a in accounts
>                 if a["username"].lower() == lowercase_username]
>             if not accounts:
>                 logger.debug((  # This would also happen when the cache is empty
>                     "get_accounts(username='{}') finds no account. "
>                     "If tokens were acquired without 'profile' scope, "
>                     "they would contain no username for filtering. "
>                     "Consider calling get_accounts(username=None) instead."
>                     ).format(username))
>         # Does not further filter by existing RTs here. It probably won't matter.
>         # Because in most cases Accounts and RTs co-exist.
>         # Even in the rare case when an RT is revoked and then removed,
>         # acquire_token_silent() would then yield no result,
>         # apps would fall back to other acquire methods. This is the standard pattern.
>         return accounts
> From the code comment, it seems clear that MSAL is "expecting" that multiple accounts can be returned, and the calling application should display a picker or something to the user for them to select which account to use.
> 
> > Your app can choose to display those information to end user, and allow user to choose one of his/her accounts to proceed.
> 
> However, this may be when the username parameter is not specified? Looking deeper in MSAL Py, the include both authority types when querying account info:
> 
> https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/dev/msal/application.py#L1254
> 
> def _find_msal_accounts(self, environment):
>         interested_authority_types = [
>             TokenCache.AuthorityType.ADFS, TokenCache.AuthorityType.MSSTS]
> So it seems like MSAL here can return the same account information that has been associated with multiple authority types?

Based on this, should Azure CLI be post-filtering the returned accounts to target the specific active cloud environment? Based on authority / issuer.

**Comment by rai69** [comment]
Same here.  Help please!
Retrieving tenants and subscriptions for the selection...
Found multiple accounts with the same username 'raimond@green-cloud.nl': [{'home_account_id': '00000000-0000-0000-6e70-74e03591ff37.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'raimond@green-cloud.nl', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': 'd64e16e4-a679-4eb4-acfc-777487e10c78', 'realm': '883b64f8-597c-4afe-b527-6409129db9ff'}, {'home_account_id': '7c386c87-05ce-4140-a2d9-80b94e9b1135.d2cd2d1c-c9e7-4bca-818b-fe307363c2a6', 'environment': 'login.microsoftonline.com', 'username': 'raimond@green-cloud.nl', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '7c386c87-05ce-4140-a2d9-80b94e9b1135', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

**Comment by JeremyTCD** [comment]
Same problem. Somehow ended up with two accounts using the same email, a personal account and a work and school account. Can't seem to close either. Been having sporadic trouble with logging in to Azure for a while now.

```
Found multiple accounts with the same username 'dev@xxxxxx': [{'home_account_id': '40cdda6b-d72d-48b4-9c26-xxxxxx.30b83047-24ac-43c7-947d-xxxxxx', 'environment': 'login.microsoftonline.com', 'username': 'dev@xxxxx', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '40cdda6b-d72d-48b4-9c26-xxxxxx', 'realm': '30b83047-24ac-43c7-947d-xxxxxx'}, {'home_account_id': '00000000-0000-0000-a7b0-xxxxxx.9188040d-6c67-4c5b-b112-xxxxxx', 'environment': 'login.microsoftonline.com', 'username': 'dev@xxxxx', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '40cdda6b-d72d-48b4-9c26-xxxxxx', 'realm': '30b83047-24ac-43c7-947d-xxxxxx'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168
```

**Comment by euangordon** [comment]
> Same problem. Somehow ended up with two accounts using the same email, a personal account and a work and school account. Can't seem to close either. Been having sporadic trouble with logging in to Azure for a while now.
> 
> ```
> Found multiple accounts with the same username 'dev@xxxxxx': [{'home_account_id': '40cdda6b-d72d-48b4-9c26-xxxxxx.30b83047-24ac-43c7-947d-xxxxxx', 'environment': 'login.microsoftonline.com', 'username': 'dev@xxxxx', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '40cdda6b-d72d-48b4-9c26-xxxxxx', 'realm': '30b83047-24ac-43c7-947d-xxxxxx'}, {'home_account_id': '00000000-0000-0000-a7b0-xxxxxx.9188040d-6c67-4c5b-b112-xxxxxx', 'environment': 'login.microsoftonline.com', 'username': 'dev@xxxxx', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '40cdda6b-d72d-48b4-9c26-xxxxxx', 'realm': '30b83047-24ac-43c7-947d-xxxxxx'}]
> Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168
> ```

@JeremyTCD Honestly, after battling with this for days, the best option is to try and get rid of one of the accounts. For me, it would not let me delete the one I wanted to, because I was somehow a Super Admin, so I ended up changing the email of that one, which it let me do. Just added 123 on the end of the email address. After that, everything worked as you would expect!

**Comment by monkut** [comment]
Same issue, told me to come here....

**Comment by Nokel81** [comment]
@euangordon Same I tried deleting my other account but during the 30 days it still reports back to `az login` as having two accounts....

However, removing `~/.azure` after deleting one of my accounts fixed this issue for me. Though I think `az login` should try and do that filtering for me

**Comment by odesuk** [comment]
the same ;( 

az login
Select the account you want to log in with. For more information on login with Azure CLI, see https://go.microsoft.com/fwlink/?linkid=2271136

Retrieving tenants and subscriptions for the selection...
Found multiple accounts with the same username 'xxx@xxx.com': [{'home_account_id': '270d9715-4bf8-48ca-99f6-7ad96a45acb9.1934d421-b4b1-4d83-9252-d57c63f42f54', 'environment': 'login.microsoftonline.com', 'username': 'xxxx@xxxx.com', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': 'c0325252-7861-48b7-a881-78adbe854c5f', 'realm': '90fc6faa-a809-4f74-bcc0-50f9daf685fd'}, {'home_account_id': '00000000-0000-0000-0435-84f612f22b45.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'xxxx@xxx.com', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '00000000-0000-0000-0435-84f612f22b45.9188040d-6c67-4c5b-b112-36a304b66dad.f8cdef31-a31e-4b4a-93e4-5f571e91255a', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

**Comment by aukolov** [comment]
Same problem.

**Comment by giang-ha-kirbycorp-com** [comment]
this fixed it for me 

Run this cmd first

`az account clear`


then run

` az login`

**Comment by YuraQlog** [comment]
thanks. fixed: deleted `.azure/msal_token_cache.json` and again `az login`

**Comment by MukkeDK** [comment]
As others have commented, this happens when you have both a personal account and a work/school account with the same login, like me@mydomain.com

While arguments can make that some things will be easier if one does not have 2 accounts like that, it is by design that this is possible. As such, az cli should support this scenario.

**Comment by AndyYouens** [comment]
> .azure/msal_token_cache.json

my file for this resides in %USERPROFILE%\.azure

**Comment by MohanRamV** [comment]
Those who of you are looking for solution, Follow the steps below:
I also faced same error, then i removed .azure folder in C:\Users\<username>
Then open CLI and type "az login --allow-no-subscriptions".
Follow the login procedure. Boom!! you are loggedin.

**Comment by synmux** [comment]
Same issue. Won't post UUIDs but the email is dave at dave dot io

**Comment by mikegorman-nf** [comment]
Was directed by CLI.

Found multiple accounts with the same username 'mike.gorman@netfoundry.io': [{'home_account_id': '00000000-0000-0000-440d-c64679bbbf7a.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'mike.gorman@netfoundry.io', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '00000000-0000-0000-440d-c64679bbbf7a.9188040d-6c67-4c5b-b112-36a304b66dad.f8cdef31-a31e-4b4a-93e4-5f571e91255a', 'realm': 'organizations'}, {'home_account_id': '52ebd09a-3737-4512-b86d-337226e5fef7.25445e86-2ae6-4434-b116-25c66c27168d', 'environment': 'login.microsoftonline.com', 'username': 'mike.gorman@netfoundry.io', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '52ebd09a-3737-4512-b86d-337226e5fef7', 'realm': '25445e86-2ae6-4434-b116-25c66c27168d'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168

I'd love to cut it down to one account from that email, but I've never figured out exactly which does what across my computers, Azure, O365, etc., so I've been very afraid to touch them.  Is there a way to review an ID of some kind to disambiguate the logins?

**Comment by michael-delphos** [comment]
Same. Was directed here by the CLI:

```
Found multiple accounts with the same username 'michael@delphoslabs.com': [{'home_account_id': 'c4047f25-b6d7-4d33-af8b-d437547ba08e.ba99b4fa-823e-4a53-992a-3b746151e901', 'environment': 'login.microsoftonline.com', 'username': 'michael@delphoslabs.com', 'account_source': 'urn:ietf:params:oauth:grant-type:device_code', 'authority_type': 'MSSTS', 'local_account_id': 'c4047f25-b6d7-4d33-af8b-d437547ba08e', 'realm': 'ba99b4fa-823e-4a53-992a-3b746151e901'}, {'home_account_id': '00000000-0000-0000-2ec8-d7b65eefb600.9188040d-6c67-4c5b-b112-36a304b66dad', 'environment': 'login.microsoftonline.com', 'username': 'michael@delphoslabs.com', 'account_source': 'broker', 'authority_type': 'MSSTS', 'local_account_id': '00000000-0000-0000-2ec8-d7b65eefb600.9188040d-6c67-4c5b-b112-36a304b66dad.f8cdef31-a31e-4b4a-93e4-5f571e91255a', 'realm': 'organizations'}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168
```

**Comment by JSGInray** [comment]
Unfortunately, none of the recommendations in this issue helped; I had to close the account and then clear the identity cache:

Navigate to the folder C:\Users<username>\AppData\Local\Microsoft\IdentityCache on your system.

Delete all files within the IdentityCache folder.

Open a command prompt or terminal.

Run the command az account clear to clear any cached Azure account information.

Re-login to Azure CLI using the command az login.

If required, specify the tenant explicitly by running az login --tenant <tenant_id>.

**Comment by AJakubowska** [comment]
Thanks @JSGInray, your solution worked for me!

**Comment by NBAlex** [comment]
I experienced the same issues. What fixed the problem for me was:

1. `az account clear`
2. `az login --tenant <tenant_id>`
3. `az account set --subscription <subscription_id>`

**Comment by orangit-vaheed-sadeghi** [comment]
same here:

**Found multiple accounts with the same username '******': [{'home_account_id': '*****', 'environment': 'login.microsoftonline.com', 'username': '******', 'account_source': 'authorization_code', 'authority_type': 'MSSTS', 'local_account_id': '*******', 'realm': 'organizations'}, {'home_account_id': '******', 'environment': 'login.microsoftonline.com', 'username': '******', 'account_source': 'authorization_code', 'authority_type': 'MSSTS', 'local_account_id': '******', 'realm': '*****}]
Please report to us via Github: https://github.com/Azure/azure-cli/issues/20168**

### **I ran `az account clear` and tried again. it fixed.**

**Comment by cnotin** [comment]
On my side I have this issue when I recycle UPNs between users.
- create user with a certain UPN
- log in with az cli with this user
- delete the user
- create a new user with same UPN
- try to log in -> error

Ok it's the same UPN but in the end it's a different user from the Entra ID point of view, with a different ID. I suppose that Azure CLI indexes something by UPN where it shouldn't.

**Comment by yangs2016** [comment]
the command **az account clear** works for me to get rid of "Found multiple accounts with same username.."

**Comment by yangs2016** [comment]
Though command  **az account clear** did help me get rid of message " 
Found multiple accounts with the same username..." when I ran "**az login**", but I still have two accounts with same username but different cases (one all lower case, one Pascal case) to choose from. Why do I have two accounts listed and how do I get rid of the one that shouldn't be there?

**Comment by Dany3R9** [comment]
az account clear 
tested today and solved my issue!

**Comment by jeremydein** [comment]
> On my side I have this issue when I recycle UPNs between users.
> 
> * create user with a certain UPN
> * log in with az cli with this user
> * delete the user
> * create a new user with same UPN
> * try to log in -> error
> 
> Ok it's the same UPN but in the end it's a different user from the Entra ID point of view, with a different ID. I suppose that Azure CLI indexes something by UPN where it shouldn't.

Mine wasn't a recycle of the UPN, but I did change the UPN of my user in Entra. As others have mentioned, `az account clear` fixed my issue.

**Comment by ivassallo19** [comment]
Also received this error today.

<img width="1919" height="239" alt="Image" src="https://github.com/user-attachments/assets/d86fa2eb-cd26-4b48-a2dd-980f5565ea85" />

az login from PowerShell 7.6.6.
Azure CLI 2.83.0 installed using MSI.
Windows 11 24H2.

Deleting \.azure\msal_token_cache.bin is what I did first and helped me login successfully.
